### PR TITLE
Add TLS between gate and service

### PR DESCRIPTION
**Pull request by davidhadas**
This PR adds TLS between guard-gate and guard-service

It supports working with a private CA or a public CA
To enable private CAs, a rootCA is loaded from env.

### CODE REVIEW

**Review by psschwei — COMMENTED** [review]
Should we consider automating the cert creation somehow?

A couple of options: using control-protocol/pkg/certificates as in https://github.com/knative/serving/pull/13005 or possibly using something like https://github.com/knative-sandbox/net-certmanager

**Comment by davidhadas** [comment]
/hold
Evaluate if we can reuse the internal ca mechanisms used by activator to approach QP

File: cmd/guard-rproxy/main.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func preMain() (guardGate *GuardGate, mux *http.ServeMux, target string, plugCon
 
 	if env.GuardUrl == "" {
 		// use default
-		env.GuardUrl = "http://guard-service.default"
+		env.GuardUrl = "http://guard-service.knative-serving"
```
Comment (psschwei) [inline cmd/guard-rproxy/main.go:100]: update comment too?

File: config/deploy/guard-service.yaml [hunk 2]
```diff
@@ -14,14 +15,27 @@ spec:
       labels:
         app: guard-service
     spec:
+#      imagePullSecrets:
+#      - name: all-icr-io
```
Comment (psschwei) [inline config/deploy/guard-service.yaml:19]:
```suggestion
```

File: cmd/guard-rproxy/main.go [hunk 3]
```diff
@@ -97,7 +97,7 @@ func preMain() (guardGate *GuardGate, mux *http.ServeMux, target string, plugCon
 
 	if env.GuardUrl == "" {
 		// use default
```
Comment (davidhadas) [inline cmd/guard-rproxy/main.go:99]:
```suggestion
```

**Review by davidhadas — COMMENTED** [review]
nits

**Review by psschwei — COMMENTED** [review]
/lgtm

**Pull request merged**